### PR TITLE
Implement the rest of TeamPrototype and more of Team

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/common/rts/team.cpp [hunk 1]
```diff
@@ -829,3 +831,698 @@ bool Team::Remove_Override_Team_Relationship(unsigned int id)
...
+}
+
+bool Team::Damage_Team_Members(float amount)
+{
+    DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+    while (!it.Done()) {
+        if (!it.Cur()->Is_Effectively_Dead()) {
+            if (!it.Cur()->Is_Destroyed()) {
+                if (amount > 0.0f) {
```
Comment (tomsons26) [inline src/game/common/rts/team.cpp]: >= or flip it and < as going by asm

Comment (jonwil) [reply src/game/common/rts/team.cpp]: Flipped and fixed.

File: src/game/common/rts/team.cpp [hunk 2]
```diff
@@ -829,3 +831,698 @@ bool Team::Remove_Override_Team_Relationship(unsigned int id)
...
+                        }
+                    }
+                }
+
+                it.Advance();
+            }
+
+            if (found_update && is_idle && m_wasIdle) {
+                g_theScriptEngine->Run_Script(info->m_scriptOnIdle, this);
+                m_wasIdle = is_idle;
```
Comment (tomsons26) [inline src/game/common/rts/team.cpp]: that line is outside this if

Comment (jonwil) [reply src/game/common/rts/team.cpp]: fixed.

File: src/game/logic/object/object.cpp [hunk 3]
```diff
@@ -367,3 +369,34 @@ void Object::Set_Team(Team *team)
...
+
+float Object::Get_Vision_Range() const
+{
+#ifdef GAME_DEBUG_STRUCTS
+    if (g_theWriteableGlobalData->m_debugVisibility) {
+        Vector3 v(m_visionRange, 0.0f, 0.0f);
+
+        for (int i = 0; i < g_theWriteableGlobalData->m_debugVisibilityTileCount; i++) {
+            float f1 = i * 1.0f / g_theWriteableGlobalData->m_debugVisibilityTileCount;
+            float angle = (f1 + f1) * 3.1415927;
```
Comment (xezon) [inline src/game/logic/object/object.cpp]: Do we have a matching PI constant for this? Also, is double, not float?

File: src/game/logic/scriptengine/script.h [hunk 4]
```diff
@@ -50,6 +50,12 @@ class Script : public MemoryPoolObject, public SnapShot
     void Set_Action(ScriptAction *action) { m_action = action; }
     void Set_False_Action(ScriptAction *action) { m_actionFalse = action; }
     void Update_Exec_Time(float passed_time) { m_totalExecTime += passed_time; }
+    int Get_UnkInt2() const { return m_unkInt2; }
+    int Get_Evaluation_Interval() const { return m_evaluationInterval; }
+    void Set_UnkInt2(int set) { m_unkInt2 = set; }
+    bool Is_Normal() { return m_normal; }
+    bool Is_Hard() { return m_hard; }
+    bool Is_Easy() { return m_easy; }
```
Comment (xezon) [inline src/game/logic/scriptengine/script.h]: const

File: src/game/common/rts/team.cpp [hunk 5]
```diff
@@ -829,3 +831,664 @@ bool Team::Remove_Override_Team_Relationship(unsigned int id)
...
+    }
+
+    if (m_productionConditionScript != nullptr) {
+        if (g_theGameLogic->Get_Frame() < m_productionConditionScript->Get_UnkInt2()) {
+            return false;
+        } else {
+            int interval = m_productionConditionScript->Get_Evaluation_Interval();
+
+            if (interval > 0) {
+                m_productionConditionScript->Set_UnkInt2(30 * interval + g_theGameLogic->Get_Frame());
```
Comment (xezon) [inline src/game/common/rts/team.cpp:1105]: `Set_UnkInt2` has no known name yet?

Comment (jonwil) [reply src/game/common/rts/team.cpp:1105]: No.

File: src/game/logic/scriptengine/script.h (hunk 4 above)
Comment (jonwil) [reply src/game/logic/scriptengine/script.h]: Original has these as const. Fixed.

File: src/game/logic/object/object.cpp (hunk 3 above)
Comment (jonwil) [reply src/game/logic/object/object.cpp]: We do. Fixed.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**